### ISSUE TITLE
TABLE DDL with IDENTITY adds unnecessary DEFAULT

### ISSUE THREAD

**Issue by Stegallo** [littleK0i/SnowDDL#27]
**Describe the bug**
I did run convert to create the yaml file of an existing table having an IDENTITY column.
The resulting yaml has `default: IDENTITY START 0 INCREMENT 1`.
When I run the apply, the resulting DDL is syntactically invalid.

Not exactly sure if the issue is in the way the column gets converted or in the way the DDL is created.
If the latter, the fix may be as simple as conditionally remove the keyword DEFAULT in 
https://github.com/littleK0i/SnowDDL/blob/master/snowddl/resolver/table.py#L170

**Expected behavior**
`DEFAULT` keyword should not be present in column definition.
```
CREATE TABLE "DB_NAME"."SCHEMA_NAME"."TABLE_NAME"
(
      "ID" NUMBER(38,0) IDENTITY START 0 INCREMENT 1 NOT NULL
)
```

**Attach log**
```
2022-11-07 15:02:23.819 - WARNING - Resolved TABLE [DB_NAME.SCHEMA_NAME.TABLE_NAME]: ERROR
(
    message   =>  SQL compilation error:
syntax error line 3 at position 47 unexpected 'START'.
    errno     =>  1003
    sqlstate  =>  42000
    sfqid     =>  01a828c6-0605-1a48-0000-6a6d07215cee
    sql       =>  CREATE TABLE "DB_NAME"."SCHEMA_NAME"."TABLE_NAME"
(
      "ID" NUMBER(38,0) DEFAULT IDENTITY START 0 INCREMENT 1 NOT NULL
)
```

**Attach YAML config (if applicable)**
```
columns:
  ID:
    type: NUMBER(38,0) NOT NULL
    default: IDENTITY START 0 INCREMENT 1
```

thank you!

**Comment by littleK0i** [comment]
Thank you for reporting this. I'll fix it next week when I get back from travel.

SnowDDL does not support `IDENTITY` for columns directly (on purpose), but we can create a config for SEQUENCE object automatically, and use this SEQUENCE object instead of IDENTITY.

**Comment by littleK0i** [comment]
@Stegallo , please check version `0.11.0`.

IDENTITY columns are now converted into sequences automatically. Current value and increment will be preserved.

Sequence name is built using the following template: `{tableName}_{colName}_seq`.

**Comment by Xiaoming-lw** [comment]
Hi @littleK0i , Thank you for your excellent work.

 I am still facing this issue while convert&apply against the ID column. It seems the snowddl-convert does not convert the IDENTITY columns into sequences. Not sure if there is any configuration need to be done. I did not find any doc and discussion related with the IDENTITY/sequence.

**Creation of the table:**
```
-- create database "CDB_A1";
-- create schema "CDB_A1"."EVENT_INTERNAL";
CREATE TABLE "CDB_A1"."EVENT_INTERNAL"."EVENT_T_1"
(
      "ID" NUMBER(38,0) IDENTITY START 1 INCREMENT 1 ORDER
)
```

**Yaml output of the snowddl-convert**
```
columns:
  id:
    type: NUMBER(38,0)
    default: IDENTITY START 1 INCREMENT 1 ORDER
```

**Errors while snowddl apply:**
```
2024-09-16 14:32:00.439 - WARNING - Resolved TABLE [CDB_A1.EVENT_INTERNAL.EVENT_T_1]: ERROR
(
    message   =>  SQL compilation error:
syntax error line 3 at position 41 unexpected 'START'.
    errno     =>  1003
    sqlstate  =>  42000
    sfqid     =>  01b713cc-0203-138f-0000-880d0df9f916
    sql       =>  CREATE TABLE "CDB_A1"."EVENT_INTERNAL"."EVENT_T_1"
(
      "ID" NUMBER(38,0) DEFAULT IDENTITY START 1 INCREMENT 1 ORDER
)
```

**snowddl version:**
```
snowddl @ git+https://github.com/littleK0i/SnowDDL@b0756c4e61de65b29a4b9b8166c3a792b3d97f16
```

**Comment by littleK0i** [comment]
@Xiaoming-lw , I guess it is a new issue. Snowflake recently introduced a concept of ordered and non ordered sequences.

So now they add "ORDER" at the end of identity definition. I'll fix in the next version.

In the meantime you may add ` ORDER` at the end of this regexp manually: https://github.com/littleK0i/SnowDDL/blob/master/snowddl/converter/table.py#L12

**Comment by Xiaoming-lw** [comment]
Hi @littleK0i , thanks for your help.

I tested with the new regex:

`x = re.search(r"^IDENTITY START (\d+) INCREMENT (\d+)( ORDER| NOORDER)?$", txt)`

The ID could be converted to sequence now, but still get error while applying.

**event_t_1.yaml:**
```
columns:
  id:
    type: NUMBER(38,0)
    default_sequence: event_t_1_id_seq
```

**event_t_1_id_seq.yaml**
```
start: 1
interval: 1
```

**apply error:**
```
2024-09-16 17:22:44.051 - WARNING - Resolved TABLE [CDB_A1.EVENT_INTERNAL.EVENT_T_1]: ERROR
(
    message   =>  SQL compilation error: error line 3 at position 32
invalid identifier 'CDB_A1.EVENT_INTERNAL.EVENT_T_1_ID_SEQ.NEXTVAL'
    errno     =>  904
    sqlstate  =>  42000
    sfqid     =>  01b71476-0203-138f-0000-880d0dfa2962
    sql       =>  CREATE TABLE "CDB_A1"."EVENT_INTERNAL"."EVENT_T_1"
(
      "ID" NUMBER(38,0) DEFAULT CDB_A1.EVENT_INTERNAL.EVENT_T_1_ID_SEQ.NEXTVAL
)
```

**Comment by littleK0i** [comment]
Conversion result looks good. You may want to add parameter `is_ordered: true` if actual ordering is important for you. Sequences are now unordered by default.

It seems sequence was not created for some reason. Maybe it is in the wrong folder or maybe `--include-object-types` is specified and does not include sequence.

Maybe you could "destroy" and re-apply everything and post the full log here. You should see sequence object being created before table object. If it does not appear in logs, something is wrong with config.

**Comment by Xiaoming-lw** [comment]
Yes, the error reason is `--include-object-types`. `SEQUENCE` was not created. Thank you @littleK0i !

**Comment by littleK0i** [comment]
Should be fixed in `0.32.0`. New flag `ORDER` / `NOORDER` is now handled correctly.
